core/src/main/java/org/elasticsearch/index/query/IndicesQueryParser.java

```diff
@@ -145,6 +148,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (queryName != null) {
             parseContext.addNamedQuery(queryName, chosenQuery);
         }
+        chosenQuery.setBoost(boost);
```

Comment:
it's a reference, there is really no difference. I am afraid we do this in a lot of other places too, but its just fine no?